modules/nf-core/ctree/templates/main_script.R

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env Rscript
+
+parse_args = function(x) {
+    x = gsub("\\\\[","",x)
+    x = gsub("\\\\]","",x)
+    # giving errors when we have lists like c(xxx, xxx) since it will separate it
+    # args_list = unlist(strsplit(x, ', ')[[1]])
+    args_list = unlist(strsplit(x, ", (?=[^)]*(?:\\\\(|\$))", perl=TRUE))
+    # args_vals = lapply(args_list, function(x) strsplit(x, split=":")[[1]])
+    args_vals = lapply(args_list, function(x) {
+        x_splt = strsplit(x, split=":")[[1]]
+        c(x_splt[1],  paste(x_splt[2:length(x_splt)], collapse=":"))
+    })
+
+    # Ensure the option vectors are length 2 (key/ value) to catch empty ones
+    args_vals = lapply(args_vals, function(z){ length(z) = 2; z})
+
+    parsed_args = structure(lapply(args_vals, function(x) x[2]), names = lapply(args_vals, function(x) x[1]))
+    parsed_args[! is.na(parsed_args)]
+}
+
+opt = list(
+    prefix = ifelse('$task.ext.prefix' == 'null', '$meta.id', '$task.ext.prefix')
+)
+args_opt = parse_args('$task.ext.args')
+for ( ao in names(args_opt)) opt[[ao]] = args_opt[[ao]]
+
+
+# Load packages
+
+library(ctree)
+library(mobster)
+library(VIBER)
+library(cli)
+library(dplyr)
+library(ggplot2)
+
+outdir = ""
+
+# Auxiliary functions
+
+add_dummy_driver = function(input_table, variant_colname, is_driver_colname) {
+    add_driver = FALSE
+    idx = dplyr::case_when(
+        "cluster" %in% colnames(input_table) ~ which(input_table[["cluster"]] != "Tail")[1],
+        .default = 1
+    )
+    if (!variant_colname %in% colnames(input_table) | !is_driver_colname %in% colnames(input_table)) {
+        input_table = input_table %>% dplyr::mutate(!!is_driver_colname:=FALSE, !!variant_colname:=NA)
+        add_driver = TRUE
+    } else if (all(input_table[[is_driver_colname]]==FALSE)) {
+        add_driver = TRUE
+    }
+    if (add_driver) {
+        input_table[idx, is_driver_colname] = TRUE
+        input_table[idx, variant_colname] = ""
+    }
+    return(input_table)
+}
+
+initialize_ctree_obj_pyclone = function(ctree_input) {
+    driver_cluster = unique(ctree_input[which(ctree_input["is.driver"]==TRUE),c("cluster")])
+    # the CCF table must report CCF values for each cluster and sample
+    # cluster | nMuts | is.driver | is.clonal | sample1 | sample2 | ...
+    CCF_table = ctree_input %>%
+        dplyr::select(sample_id, cluster, nMuts, is.driver, is.clonal, CCF) %>%
+        dplyr::mutate(is.driver=replace(is.driver, is.driver=="", "FALSE")) %>%
+        dplyr::mutate(is.driver=as.logical(is.driver)) %>%
+        dplyr::filter(cluster!="Tail") %>%
+        dplyr::mutate(cluster=as.character(cluster)) %>%
```

Comment:
Surely it is already a character vector, or the previous line wouldn't be able to filter?
You can group_by a factor or numeric, doesn't have to be a character.